client/Windows/wf_cliprdr.c

```diff
@@ -2191,14 +2191,21 @@ wf_cliprdr_server_format_data_response(CliprdrClientContext* context,
 	if (!context || !formatDataResponse)
 		return ERROR_INTERNAL_ERROR;
 
-	if (formatDataResponse->common.msgFlags != CB_RESPONSE_OK)
-		return E_FAIL;
-
 	clipboard = (wfClipboard*)context->custom;
 
 	if (!clipboard)
 		return ERROR_INTERNAL_ERROR;
 
+	if (formatDataResponse->common.msgFlags != CB_RESPONSE_OK)
+	{
+		clipboard->hmem = NULL;
```

Comment:
Hm - sure we don't leak any memory here?

Reply:
@mfleisz no, this should already be handled. just look a few lines below, where it normally is allocated.

Reply:
@akallabeth a few lines below the newly allocated memory is assigned to `clipboard->hmem` (line 2230) and if we just `NULL` it here that handle will be lost forever ... or do I miss something else?

Reply:
@mfleisz yes, the `return CHANNEL_RC_OK` in the new path?